brayns/common/scene/Scene.cpp

```diff
@@ -669,4 +675,325 @@ void Scene::setCylinder(const size_t materialId, const uint64_t index,
     else
         BRAYNS_ERROR << "Invalid index " << index << std::endl;
 }
+
+void Scene::saveToCacheFile()
+{
+    const auto& filename =
+        _parametersManager.getGeometryParameters().getSaveCacheFile();
+    BRAYNS_INFO << "Saving scene to binary file: " << filename << std::endl;
+    std::ofstream file(filename, std::ios::out | std::ios::binary);
+    if (!file.good())
+    {
+        BRAYNS_ERROR << "Could not open cache file " << filename << std::endl;
+        return;
+    }
+
+    const size_t version = CACHE_VERSION;
+    file.write((char*)&version, sizeof(size_t));
```

Comment:
why not file << version? everywhere in this file?

Reply:
Not all elements have this operator implemented. Some of those elements being in process of being replaced, I will leave your comments to a later PR if that's acceptable?